node/lib/util/push.js

```diff
@@ -93,24 +94,23 @@ exports.push = co.wrap(function *(repo, remoteName, source, target, force) {
     // First, push the submodules.
 
     let errorMessage = "";
-    const subRepos = yield SubmoduleUtil.getSubmoduleRepos(repo);
     const shas = yield SubmoduleUtil.getSubmoduleShasForBranch(repo, source);
-    const index = yield repo.index();
-    const urls = yield SubmoduleConfigUtil.getSubmodulesFromIndex(repo, index);
```

Comment:
I think the strategy espoused by the Git developers may make more sense for submodules as they were intended to be used than as we are using them to build a mono-repo (though the whole story surrounding them is a little confused from the beginning).